packages/next-intl/src/navigation/react-client/useBasePathname.tsx

```diff
@@ -34,11 +35,20 @@ export default function useBasePathname<
   return useMemo(() => {
     if (!pathname) return pathname;
 
+    let unlocalizedPathname = pathname;
+
     const prefix = getLocalePrefix(locale, localePrefix);
     const isPathnamePrefixed = hasPathnamePrefixed(prefix, pathname);
-    const unlocalizedPathname = isPathnamePrefixed
-      ? unprefixPathname(pathname, prefix)
-      : pathname;
+
+    if (isPathnamePrefixed) {
+      unlocalizedPathname = unprefixPathname(pathname, prefix);
+    } else {
+      // https://github.com/vercel/next.js/issues/73085
+      const localeAsPrefix = getLocaleAsPrefix(locale);
+      if (hasPathnamePrefixed(localeAsPrefix, pathname)) {
+        unlocalizedPathname = unprefixPathname(pathname, localeAsPrefix);
+      }
+    }
```

Comment:
One rare case where this behavior might not be wanted is this:

```tsx
const routing = defineRouting({
  locales: ['en'],
  defaultLocale: 'en',
  localePrefix: {
    mode: 'as-need', // Anything other than "always"
    prefixes: {
      en: '/uk'
    }
  }
});
```

… and then having a path like `/[locale]/en/page.tsx`.

In this case, when Next.js prerenders with a pathname set to `/en/en`, we end up with `/en` as the unprefixed pathname—that's fine. However, in production, the Next.js pathname will be `/en` after the rewrite and we'll unprefix this to `/`.

A potential workaround could be based on `useParams()`, `useSelectedLayoutSegments()` etc., but I believe a static path like `/en` can still not be detected due to only the `locale` segment being returned from these hooks. So what would be necessary is a hook that returns the actual, internal pathname, but Next.js currently doesn't support that.

So I think for the time being this is a tradeoff that we have to live with.